### PR TITLE
HDF5 qmmm checkpoint reader

### PR DESCRIPTION
**Pull request by JensWehner**
jobtopology can read hdf5 files now. 

Some small changes using more std::algorithm and some other small refactors.

### CODE REVIEW

**Comment by JensWehner** [comment]
@votca-bot format

**Comment by JensWehner** [comment]
@votca-bot changelog: added a checkpoint reader to jobtopology class

**Comment by JensWehner** [comment]
@votca-bot format

File: src/libxtp/jobcalculators/qmmm.cc [hunk 1]
```diff
@@ -133,10 +139,10 @@ Job::JobResult QMMM::EvalJob(const Topology& top, Job& job, QMThread& Thread) {
...
       }
 
-      double etot = 0.0;
-      for (const std::unique_ptr<Region>& reg : jobtop) {
-        etot += reg->Etotal();
-      }
+      double etot = std::accumulate(
+          jobtop.begin(), jobtop.end(), 0.0,
+          [](double e, const auto& reg) { return e + reg->Etotal(); });
+
```
Comment (felipeZ) [inline src/libxtp/jobcalculators/qmmm.cc:145]: So much better!

File: src/libxtp/jobtopology.cc [hunk 2]
```diff
@@ -349,5 +349,46 @@ void JobTopology::WriteToHdf5(std::string filename) const {
...
+  CheckpointFile cpf(filename, CheckpointAccessLevel::READ);
+  CheckpointReader a = cpf.getReader();
+  Index id = -1;
+  a(id, "jobid");
+  if (id != _job.getId()) {
+    throw std::runtime_error(
+        "Jobid from checkpoint file does not agree with jobid" +
+        std::to_string(id) + ":" + std::to_string(_job.getId()));
+  }
+  _regions.clear();
```
Comment (felipeZ) [inline src/libxtp/jobtopology.cc:362]: Out of curiosity, why do you call this `clear` here?

**Review by felipeZ — APPROVED** [review]

File: src/libxtp/jobtopology.cc (hunk 2 above)
Comment (JensWehner) [reply src/libxtp/jobtopology.cc:362]: otherwise it could in principle append the regions from the file, so I delete all the old regions first.

**Pull request merged**